src/cache/surface_albedo.jl

```diff
@@ -0,0 +1,204 @@
+
+A constant surface albedo model. The default value is 0.38. It is used purely for idealized experiments.
+"""
+Base.@kwdef struct ConstantAlbedo{FT} <: SurfaceAlbedoModel
+    α::FT = 0.38
+end
+
+"""
+    struct RegressionFunctionAlbedo{FT} <: SurfaceAlbedoModel
+
+A regression function-based surface albedo model of Jin et al. (2011, referred to as J11).
+"""
+struct RegressionFunctionAlbedo{FT, F <: Function} <: SurfaceAlbedoModel
+    n::FT                           # relative refractive index of water and air (n = n_w/n_a) TODO: f(wavelength) for a spectrally dependent scheme
+    n0::FT                          # refractive index of water for visible light
+    p::SA.SVector{11, FT}           # regression coefficients of J11 for eq. 4
+    q_clear::SA.SVector{4, FT}      # regression coefficients of J11 for eq. 5a
+    q_cloud::SA.SVector{3, FT}      # regression coefficients of J11 for eq. 5b
+    wave_slope::F                   # mean wave slope distribution width (Cox-Munk model, J11 Eq. 2)
+end
+function RegressionFunctionAlbedo{FT}(;
+    n = FT(1.34),
+    n0 = FT(1.34),
+    p = SA.SVector{11, FT}(
+        0.0152,
+        -1.7873,
+        6.8972,
+        -8.5778,
+        4.071,
+        -7.6446,
+        0.1643,
+        -7.8409,
+        -3.5639,
+        -2.3588,
+        10.0538,
+    ),
+    q_clear = SA.SVector{4, FT}(-0.1482, -0.012, 0.1608, -0.0244),
+    q_cloud = SA.SVector{3, FT}(-0.1479, 0.1502, -0.016),
+    wave_slope = u -> sqrt(FT(0.003) + FT(0.00512) * u),
+) where {FT <: AbstractFloat}
+    return RegressionFunctionAlbedo(n, n0, p, q_clear, q_cloud, wave_slope)
+end
+
+"""
+    set_surface_albedo!(Y, p, t, α_model::ConstantAlbedo)
+
+Set the surface albedo to a constant value.
+"""
+function set_surface_albedo!(Y, p, t, α_model::ConstantAlbedo{FT}) where {FT}
+
+    (; direct_sw_surface_albedo, diffuse_sw_surface_albedo) =
+        p.radiation.radiation_model
+
+    @. direct_sw_surface_albedo = α_model.α
+    @. diffuse_sw_surface_albedo = α_model.α
+end
+
+"""
+    set_surface_albedo!(Y, p, t, α_model::RegressionFunctionAlbedo{FT})
+
+Wrapper to call the formulations of Jin et al. (2011) for direct and diffuse surface albedos of the ocean.
+"""
+function set_surface_albedo!(
+    Y,
+    p,
+    t,
+    α_model::RegressionFunctionAlbedo{FT},
+) where {FT}
+
+    (; direct_sw_surface_albedo, diffuse_sw_surface_albedo, cos_zenith) =
+        p.radiation.radiation_model
+
+    λ = FT(0) # spectral wavelength (not used for now)
+    μ = cos_zenith
+
+    u_int = Fields.field_values(Fields.level(Y.c.uₕ, 1))
+    int_local_geometry_values =
+        Fields.field_values(Fields.level(Fields.local_geometry_field(Y.c), 1))
+
+    wind_speed = p.scratch.ᶠtemp_field_level
+    wind_speed_values = Fields.field_values(wind_speed)
+    @. wind_speed_values =
+        Geometry._norm.(
+            CT12.(u_int, int_local_geometry_values),
+            int_local_geometry_values,
+        )
```

Comment:
Would `norm(Fields.level(Y.c.uₕ, 1))` work? like [here](https://github.com/CliMA/ClimaAtmos.jl/blob/b9e5d1af9c909af361b14b948311a47688798f59/src/cache/precomputed_quantities.jl#L556)

Reply:
Ahh, that's the norm I was looking for! :D Still, I think we'll want to extrapolate to the surface as mentioned above, but this should do for now.